editor/editor_settings.cpp

```diff
@@ -1144,6 +1149,38 @@ void EditorSettings::setup_language() {
 	load_extractable_translations(lang);
 }
 
+String EditorSettings::get_online_docs_url() {
+	String language = EditorSettings::get_singleton()->get_setting("interface/editor/editor_language");
+	bool override = EditorSettings::get_singleton()->get_setting("interface/editor/override_online_docs_language");
+	String override_language = EditorSettings::get_singleton()->get_setting("interface/editor/online_docs_language");
+
+	Vector<String> doc_languages;
+	doc_languages.push_back("en");
+	doc_languages.push_back("cs");
+	doc_languages.push_back("de");
+	doc_languages.push_back("es");
+	doc_languages.push_back("fr");
+	doc_languages.push_back("it");
+	doc_languages.push_back("ja");
+	doc_languages.push_back("ko");
+	doc_languages.push_back("pl");
+	doc_languages.push_back("pt-br");
+	doc_languages.push_back("ru");
+	doc_languages.push_back("uk");
+	doc_languages.push_back("zh-cn");
+	doc_languages.push_back("zh-tw");
+
+	if (override && (doc_languages.find(override_language) > 0)) {
+		language = override_language;
+	} else if (!(doc_languages.find(language) > 0)) {
+		language = "en";
+	}
+
+	// The translated online docs only use version "4.x", not "4.3" or "latest" or "stable".
+	String version = (language == "en") ? VERSION_DOCS_BRANCH : "4.x";
```

Comment:
This should probably use the `VERSION_MAJOR` define from [`version.h`](https://github.com/godotengine/godot/blob/master/core/version.h) instead of hard-coding "4.x" here. Otherwise, for any major update, this would be a place the devs need to update the version number by hand.

Reply:
Using a different form for the translated docs may be unnecessary.
Test cases:
https://docs.godotengine.org/fr/stable/ redirects to https://docs.godotengine.org/fr/4.x/, a nice landing page.
https://docs.godotengine.org/fr/stable is a page not found.
https://docs.godotengine.org/fr/latest/ redirects to https://docs.godotengine.org/fr/4.x/.
https://docs.godotengine.org/fr/latest is a page not found.
https://docs.godotengine.org/fr/latest/classes/class_node.html redirects to https://docs.godotengine.org/fr/4.x/classes/class_node.html

So if we make sure the trailing slash is included in any editor links to the docs, and all the docs links are either `stable` or `latest` for the version, it seems to work fine. I think I did my earlier manual testing not checking whether the trailing slash was included.

Reply:
Ah, but the stable version of godot still links to `4.3`, not to `stable`, of course. So we do need the manual cases.